runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -240,6 +282,42 @@ void registerTable(@Nonnull String namespace,
         }
     }
 
+    /**
+     * For each protobuf filename -> descriptor map, validate if the previous descriptor matches
+     * If there is a new schema or an update, record the new update in the table.
+     *
+     * @param protoName - name of the protobuf file
+     * @param newProtoFd - descriptor of the protobuf file
+     * @param namespace - namespace of the table
+     * @param tableName - tablename without the namespace
+     * @return - true if schema change is detected, false otherwise
+     */
+    private boolean recordNewSchema(ProtobufFileName protoName,
+                                    CorfuRecord<ProtobufFileDescriptor, TableMetadata> newProtoFd,
+                                    String namespace,
+                                    String tableName) {
+        final CorfuRecord<ProtobufFileDescriptor, TableMetadata> oldProto = this.protobufDescriptorTable.get(protoName);
+        if (oldProto == null) {
+            this.protobufDescriptorTable.put(protoName, newProtoFd);
+            return false;
+        }
+        if (!oldProto.getPayload().getFileDescriptor().equals(newProtoFd.getPayload().getFileDescriptor())) {
```

Comment:
Yes, I thought about this, but this requires some form of safe garbage collection - a sweep of the table registry and descriptor table to delete non-referenced stale entries. This would need to be done as part of compaction probably for correctness.